### PR TITLE
⚠️ Remove default resource limits

### PR DESCRIPTION
**Pull request by axel7born**
⚠️ Operators: you might want to configure resource limits of the extension deployment explicitly in your `ControllerDeployment`.

We propose to set the resources field in the values.yaml file to an empty map. This change would align with the configuration used in other [Gardener extensions](https://github.com/gardener/gardener-extension-provider-aws/blob/f1b5a470140aa9953e032e3967a30e91a995cdd6/charts/gardener-extension-provider-aws/values.yaml#L14). We've encountered issues specifically related to the CPU limit, which we prefer not to set. With the current configuration, it's not possible to delete the CPU limit once it's set; we can only increase its value. By setting resources to an empty map, we can avoid this issue and provide more flexibility in resource configuration.

### CODE REVIEW

File: charts/gardener-extension-acl/values.yaml [hunk 1]
```diff
@@ -3,13 +3,7 @@ imagePullPolicy: IfNotPresent
...
 replicaCount: 2
 
-resources:
-  requests:
-    cpu: "50m"
-    memory: "64Mi"
-  limits:
-    cpu: "100m"
-    memory: "128Mi"
+resources: {}
```
Comment (timebertt) [inline charts/gardener-extension-acl/values.yaml:6]: I get the motivation for removing the limits. The option to not use limits if desired makes sense to me.
I'm wondering why you chose to remove the default requests though. VPA is enabled by default but doesn't work without requests set.
I would expect that it makes most sense to have some sensible default requests.

**Review by timebertt — COMMENTED** [review]
I'm generally fine with the change, I'm just wondering about the requests part.
I marked the PR with a ⚠️ so that operators know they might want to adapt their configuration.

File: charts/gardener-extension-acl/values.yaml (hunk 1 above)
Comment (axel7born) [reply charts/gardener-extension-acl/values.yaml:6]: The only argument to set resources to an empty map is that it's consistent to other extensions.
But I agree, that it makes sense to have defaults here.
I will add the requests again.

**Review by timebertt — APPROVED** [review]
Great, thanks!

**Pull request merged**